### ISSUE TITLE
ProfilingPlugin errors when user is loading JSON. 

### ISSUE THREAD

**Issue by TheLarkInn** [webpack/webpack#6414]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
If you use the `new webpack.debug.ProfilingPlugin` available in v4, and you are loading/importing JSON via webpack (without loader, default behavior), then the plugin will throw an error. 
**If the current behavior is a bug, please provide the steps to reproduce.**

**Error message**
```
/mnt/d/FastCode/foo/node_modules/webpack/lib/debug/ProfilingPlugin.js:267
        Object.keys(instance.hooks).forEach(hookName => {
               ^

TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at interceptAllHooksFor (/mnt/d/FastCode/foo/node_modules/webpack/lib/debug/ProfilingPlugin.js:267:9)
    at moduleFactory.hooks.parser.for.tap (/mnt/d/FastCode/foo/node_modules/webpack/lib/debug/ProfilingPlugin.js:283:4)
    at SyncHook.eval [as call] (eval at create (/mnt/d/FastCode/foo/node_modules/tapable/lib/HookCodeFactory.js:17:12), <anonymous>:7:1)
    at SyncHook.lazyCompileHook [as _call] (/mnt/d/FastCode/foo/node_modules/tapable/lib/Hook.js:35:21)
    at NormalModuleFactory.createParser (/mnt/d/FastCode/foo/node_modules/webpack/lib/NormalModuleFactory.js:343:31)
    at NormalModuleFactory.getParser (/mnt/d/FastCode/foo/node_modules/webpack/lib/NormalModuleFactory.js:335:41)
    at process.nextTick (/mnt/d/FastCode/foo/node_modules/webpack/lib/NormalModuleFactory.js:250:21)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

**What is the expected behavior?**
That the plugin doesn't throw an error (in this case it is because it is trying to access `Object.keys(instance.hooks)` when instance doesn't have a hooks property. 

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
webpack@4.0.0-beta.0

## To Fix
- [ ] Protect ProfilingPlugin from throwing in these scenarios. In this case, one can wrap a simple `Reflect.has` around the instance (in the previous function on the stack) and check to see if the instance has a hooks property. If not, we should not call into `interceptAllHooksFor`. (Help wanted, will merge this work). 

## First Time Contributor Notes: 
- [ ] When you fix this add a test that backs this guard. In this scenario you can ( in your test [see other tests for this plugin], you can simply import a .json file into your entrypoint ) Since the JSONParser doesn't have this.hooks={} implemented, its a good case. You could add another Parser or instance that doesn't have this.hooks, but I think for now we are good
- [ ] This is a fix to webpack v4, therefore you should fork the `next` branch when you create your bugfix branch to fix this. 
- [ ] This is a fix to webpack v4, therefore your PR should be against `next` branch not `master`.
- [ ] Ask any questions that aren't clear as you go. This should be a fun experience.

**Comment by mmckeaveney** [comment]
I can pick this :)

**Comment by TheLarkInn** [comment]
Look forward to the PR :-D feel free and ping me on it. Also double check that Reflect.has() is supported on our min version req for node ( I believe it is, but good things for you to check anyways as a first PR ). 
@ mention me once you have the PR submitted.

**Comment by mmckeaveney** [comment]
No problem, thanks @TheLarkInn. Looks like `Reflect.has` isn't supported in node 5, which webpack should still support in v4, right? Will get a look at the code tonight, but assuming `hooks` exists on the `instance` itself and not up the prototype chain, I think we could get away with `Object.hasOwnProperty`

**Comment by mmckeaveney** [comment]
@TheLarkInn PR is up for review. Thanks for the help on this one

**Comment by TheLarkInn** [comment]
webpack/webpack's (#next branch) minimum supported node version is 6. So you can use Reflect in that case. I'll look at the PR though.

**Comment by TheLarkInn** [comment]
Fixed with #6423
